atlas/readme.md

```diff
@@ -0,0 +1,325 @@
+
+```
+atlas::lyrical.geniusSong/spotify.Song=genius.Song
+```
+
+Why? Because atlases control access by granting users the ability to write different parts of the namespace. I, as the author of lyrical, have the ability to write to the `lyrical` namespace, but not to the `spotify` namespace. Therefore, the edges I declare must be within my namespace, even if they operate on another type.
+
+#### URLs end with the layer
+
+```
+atlas::schema.Type?path=file://some/wkspce/schema.star
+atlas::schema.Type?path=file://some/wkspce/schema.star&ver=2
+```
+
+Layers are described in more detail in the [next section](#atlas-layers).
+
+### Layers stack to form the atlas
+
+When accessed from some particular client, atlas data is likely strewn across various places. We call these *layers*.
+
+For example, when compiling types against the public atlas, the layers might include:
+  - schema files in the workspace
+  - the user's global cache in ~/.atlas
+  - the public atlas in heaven
+
+In another instance, we might be using the atlas within a language server. The LSP provides access to unsaved content in the user's editor, which we can represent as another layer within the atlas.
+
+For many purposes, it's not important to distinguish between different layers of an atlas. But in other cases, it might—for instance, if the user's editor shows a diff between two versions of a schema file, we'll need different atlas URLs to identify each of them in order to parse and link them separately and show the appropriate language annotations for each.
+
+### Schema
+
+Details TK, the atlas provides [a schema](./primer/atlas.star) very much like the introspection schema, but extended with:
+  - the ability to track directives and where they're applied
+  - a `URL` scalar which holds an atlas url
+  - `id: URL` fields on all schema elements
+  - more `Query` methods
+
+Additional schemas can compose with the atlas core schema, providing queries by e.g. fileystem path and token position, or the ability to provide announcements
+
+## Filesystem binding
+
+### References resolve via scoped lookup
+
+Consider [this gql]('./fixtures/basic/admin.gql):
+
+```graphql
+extend type User @from(type: "users.User") {
+  role: Role
+}
+```
+
+Or the equivalent starql:
+
+```graphql
+extend type users.User {
+  role: Role
+}
+```
+
+In either case, to complete the type `User`, the compiler needs to resolve `users.User`. This happens in two steps:
+
+1. first, we look up the first part of the dotted path—`user`, by scanning up the [scope chain](#scopes-mirror-the-file-tree) until a scope is found in which the name `user` has been defined.
+2. then, we descend into that definition, looking for a `User` element.
+
+### Scopes mirror the file tree
+
+The scope chain consists of:
```

Comment:
Do you have a preference on if tooling defaults to local install (i.e. `./.atlas/cache`) or global (`~/.atlas/cache`)?